src/viper/README.md

```diff
@@ -186,42 +286,118 @@ Below, we summarize the language features that _Motoko-san_ currently supports.
         * Immutable: `let y = ...`
         * Fields may _not_ be initialized via block expressions: `let z = { ... };`
 
-    * **Functions** — Only functions of `()` and `async ()` type with no arguments are supported:
+    * **Functions** — Functions of multiple arguments are supported:
 
-        `public shared func claim() : async () = { ... };`
+        ```motoko
+        // public func, the result is async
+        public func f(a: Int, b: Bool) : async [Bool] { ... };
 
-        `private func reward() : () = { ... };`
+        // private func, no async
+        func g(a: Int, b: Bool) : [Bool] { ... };
+        ```
 
-        Supporting function arguments and return values is _simple_.
+        If a function result type is non-`()`, it must be returned using the
+        `return` statement.
 
-    * **Local declarations** — Only local variable declarations with trivial left-hand side are supported:
+        Polymorphism is supported in private functions only:
 
-        `var x = ...;` and `let y = ...;`
+        ```motoko
+        func firstValue<T, U>(a : T, _b : U) : T { ... };
+        ```
 
-        Supporting pattern matching declarations (e.g., `let (a, b) = ...;`) is _simple_.
+        In the Viper translation, a polymorphic function is treated as a
+        template: each instantiation of its type parameters creates a new copy
+        of the function definition, where the parameters have been instantiated
+        to concrete types. The call sites are modified to use the monomorphised
+        version. The instantiations are found by traversing the static call
+        graph within the actor.
+
+    * **Local declarations** — Local variables are declared with `var` or `let`:
+
+        ```motoko
+        var x = ...;
+        let y = ...;
+        ```
+
+        Furthermore, `let`-declarations may pattern match on a tuple:
+
+        ```motoko
+        let (a, b) = ...;
+        ```
 
 * **Statements**
 
     * `()`-typed block statements and sequential composition:
 
-        `{ var x = 0 : Int; x := x + 1; }`
+        ```motoko
+        {
+            var x = 0 : Int;
+            x := x + 1;
+        }
+        ```
 
         Supporting `let y = do { let y = 1 : Int; y + y };` is _simple_.
 
     * Runtime assertions: `assert i <= MAX;`
 
-    * Assignments (to local variables and actor fields): `x := x + 1`
-    
+    * Assignments (to local variables, actor fields, and array elements):
+
+      ```motoko
+      x := x + 1;         // x   is a local variable
+      fld := fld + 1;     // fld is an actor field
+      a[i] := a[i] + 1;   // a   is an array
+      ```
+
     * `if-[else]` statements
-  
+
         Supporting pattern-matching is conceptually _simple_.
```

Comment:
We could perhaps remove this line now

Reply:
We would first have to add support for nested pattern matching, I imagine?

Reply:
We can take it into account when we will update README after the 3rd Milestone